third_party_integration/huggingface_transformers/BootstrapNAS_HF/0001-Modifications-for-BootstrapNAS-usage.patch

```diff
@@ -0,0 +1,1918 @@
+index d27fc5fa3..885a6925c 100644
+--- a/src/transformers/models/roberta/modeling_roberta.py
++++ b/src/transformers/models/roberta/modeling_roberta.py
+@@ -179,7 +179,7 @@ class RobertaSelfAttention(nn.Module):
+         self.is_decoder = config.is_decoder
+ 
+     def transpose_for_scores(self, x: torch.Tensor) -> torch.Tensor:
+-        new_x_shape = x.size()[:-1] + (self.num_attention_heads, self.attention_head_size)
++        new_x_shape = x.size()[:-1] + (self.num_attention_heads, -1)
+         x = x.view(new_x_shape)
+         return x.permute(0, 2, 1, 3)
+ 
+@@ -268,7 +268,7 @@ class RobertaSelfAttention(nn.Module):
+         context_layer = torch.matmul(attention_probs, value_layer)
+ 
+         context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+-        new_context_layer_shape = context_layer.size()[:-2] + (self.all_head_size,)
++        new_context_layer_shape = context_layer.size()[:-2] + (-1,)
+         context_layer = context_layer.view(new_context_layer_shape)
+ 
+         outputs = (context_layer, attention_probs) if output_attentions else (context_layer,)
+diff --git a/src/transformers/optimization.py b/src/transformers/optimization.py
+index b957acb6d..eba0ba733 100644
+--- a/src/transformers/optimization.py
++++ b/src/transformers/optimization.py
+@@ -22,6 +22,7 @@ import torch
+ from torch import nn
+ from torch.optim import Optimizer
+ from torch.optim.lr_scheduler import LambdaLR
++from nncf.experimental.torch.nas.bootstrapNAS.training.stage_descriptor import StageDescriptor
+ 
+ from .trainer_utils import SchedulerType
+ from .utils import logging
+@@ -31,6 +32,27 @@ from .utils.versions import require_version
+ logger = logging.get_logger(__name__)
+ 
+ 
++class BNASLambdaLR(LambdaLR):
++    def __init__(self, optimizer: Optimizer, lr_lambda: Callable[[int], float],
++                 last_epoch: int = -1, verbose: bool = False):
++        self._current_step = -1
++        super().__init__(optimizer, lr_lambda, last_epoch, verbose)
++
++    @property
++    def current_step(self) -> int:
++        return self._current_step
++
++    def step(self, epoch: int = None) -> None:
++        super().step(epoch)
++        self._current_step += 1
++
++    def stage_step(self, stage_desc: StageDescriptor) -> None:
++        pass
++
++    def epoch_step(self, next_epoch: Optional[int] = None) -> None:
++        pass
++
++
+ def get_constant_schedule(optimizer: Optimizer, last_epoch: int = -1):
+     """
+     Create a schedule with a constant learning rate, using the learning rate set in optimizer.
+@@ -44,7 +66,7 @@ def get_constant_schedule(optimizer: Optimizer, last_epoch: int = -1):
+     Return:
+         `torch.optim.lr_scheduler.LambdaLR` with the appropriate schedule.
+     """
+-    return LambdaLR(optimizer, lambda _: 1, last_epoch=last_epoch)
++    return BNASLambdaLR(optimizer, lambda _: 1, last_epoch=last_epoch)
```

Comment:
Will it affect previous user scenarios with `LambdaLR`? 
I'd assume that patching or integration to Optimum shouldn't break existing pipelines.